src/main/java/seedu/duke/parser/Parser.java

```
@@ -22,37 +25,48 @@
         String[] inputArray = null;
         String command = null;
         String argument = null;
-        try {
-            inputArray = splitTextIntoTwoFields(userInput);
-            command = inputArray[0];
+        userInput.trim();
+
+        inputArray = splitTextIntoTwoFields(userInput);
+        command = inputArray[0];
+        if (inputArray.length != 1) {
             argument = inputArray[1];
-        } catch (Exception e) {
-            System.out.println(":)");
+        } else if (!command.equals("list") && !command.equals("exit")) {
+            throw new InvalidCommandException();
         }
-        switch (command) {
-        case CheckInCommand.COMMAND:
-            return parseCheckIn(argument);
-        case CheckoutCommand.COMMAND:
-            return parseCheckOut(argument);
-        case FindCommand.COMMAND:
-            return parseFind(argument);
-        case ListCommand.COMMAND:
-            return parseList();
-        case DeleteCommand.COMMAND:
-            return parseDelete();
-        case ExitCommand.COMMAND:
-            return parseExit();
-        default:
-            return null;
+        try {
+            switch (command) {
+            case CheckInCommand.COMMAND:
+                return parseCheckIn(argument);
+            case CheckoutCommand.COMMAND:
+                return parseCheckOut(argument);
+            case FindCommand.COMMAND:
+                return parseFind(argument);
+            case ListCommand.COMMAND:
+                return parseList();
+            case DeleteCommand.COMMAND:
+                return parseDelete();
+            case ExitCommand.COMMAND:
+                return parseExit();
+            default:
+                throw new InvalidCommandException();
+            }
+        } catch (NoArgumentPassedException e) {
+            System.out.print("No argument passed! Try again!");
+        } catch (WrongFlagException e) {
```

Cannot have empty catch statement